sdk/src/main/scala/io/horizen/account/state/ForgerStakeStorageV3.scala

```diff
@@ -0,0 +1,404 @@
+    val fromEpochNumber = r.getInt()
+    val stakeAmountLength = r.getInt()
+    val stakeAmount = new BigIntegerUInt256(r.getBytes(stakeAmountLength)).getBigInt
+    StakeCheckpoint(fromEpochNumber, stakeAmount)
+  }
+}
+
+object ForgerMap {
+  val forgerList: StateDbArray = new StateDbArray(ACCOUNT, "Forgers".getBytes("UTF-8"))
+
+  def existsForger(view: BaseAccountStateView, forgerKey: Array[Byte]): Boolean = {
+    val forgerData = view.getAccountStorage(ACCOUNT, forgerKey)
+    // getting a not existing key from state DB using RAW strategy
+    // gives an array of 32 bytes filled with 0, while using CHUNK strategy
+    // gives an empty array instead
+    !forgerData.sameElements(NULL_HEX_STRING_32)
+  }
+
+  def addForger(view: BaseAccountStateView,
+                forgerKey: Array[Byte],
+                blockSignProposition: PublicKey25519Proposition,
+                vrfPublicKey: VrfPublicKey,
+                rewardShare: Int,
+                rewardAddress: Address,
+               ): Unit = {
+    forgerList.append(view, forgerKey)
+
+    val forgerStakeData = ForgerInfoV3(
+      ForgerPublicKeys(blockSignProposition, vrfPublicKey), rewardShare, new AddressProposition(rewardAddress))
+
+    // store the forger stake data
+    view.updateAccountStorageBytes(ACCOUNT, forgerKey,
+      ForgerInfoV3Serializer.toBytes(forgerStakeData))
+  }
+
+  def updateForger(view: BaseAccountStateView,
+                forgerKey: Array[Byte],
+                blockSignProposition: PublicKey25519Proposition,
+                vrfPublicKey: VrfPublicKey,
+                rewardShare: Int,
+                rewardAddress: Address,
+               ): Unit = {
+
+    val forgerStakeData = ForgerInfoV3(
+      ForgerPublicKeys(blockSignProposition, vrfPublicKey), rewardShare, new AddressProposition(rewardAddress))
+
+    // store the forger stake data
+    view.updateAccountStorageBytes(ACCOUNT, forgerKey,
+      ForgerInfoV3Serializer.toBytes(forgerStakeData))
+  }
+
+
+  def getSize(view: BaseAccountStateView): Int = forgerList.getSize(view)
+
+  def getKeys(view: BaseAccountStateView): Seq[Array[Byte]] = {
+    val listSize = getSize(view)
+    (0 until listSize).map(index => {
+     forgerList.getValue(view, index)
+    })
+  }
+
+  def getForger(view: BaseAccountStateView, forgerKey: Array[Byte]): Option[ForgerInfoV3] = {
+    val forgerData = view.getAccountStorage(ACCOUNT, forgerKey)
+    if (!forgerData.sameElements(NULL_HEX_STRING_32))
+      Some(ForgerInfoV3Serializer.parseBytes(view.getAccountStorageBytes(ACCOUNT, forgerKey)))
+    else
+      None
+  }
+
+  def apply(view: BaseAccountStateView, forgerKey: Array[Byte]): ForgerInfoV3 = {
+    val forgerData = view.getAccountStorage(ACCOUNT, forgerKey)
+    ForgerInfoV3Serializer.parseBytes(view.getAccountStorageBytes(ACCOUNT, forgerKey))
+  }
+
+
+  def getPagedListOfForgers(view: BaseAccountStateView, startPos: Int, pageSize: Int): (Int, Seq[ForgerInfoV3]) = {
```

Comment:
Let's add a case class for this response - something like
`case class PagedForgersListResponse(endPosition: Int, forgers: Seq[ForgerInfoV3])
`

Reply:
Done